centreon/src/Core/Notification/Application/UseCase/AddNotification/AddNotification.php

```diff
@@ -0,0 +1,424 @@
+     */
+    private function addResources(int $notificationId, array $newResources): void
+    {
+        foreach ($newResources as $resourceType => $newResource) {
+            $repository = $this->resourceRepositoryProvider->getRepository($resourceType);
+
+            $repository->add($notificationId, $newResource);
+        }
+    }
+
+    /**
+     * @param int $notificationId
+     *
+     * @return NotificationResource[]
+     */
+    private function findResourcesByNotificationId(int $notificationId): array
+    {
+        $resources = [];
+        foreach ($this->resourceRepositoryProvider->getRepositories() as $repository) {
+            if ($this->user->isAdmin()) {
+                $resource = $repository->findByNotificationId($notificationId);
+            } else {
+                $accessGroups = $this->readAccessGroupRepository->findByContact($this->user);
+                $resource = $repository->findByNotificationIdAndAccessGroups($notificationId, $accessGroups);
+            }
+
+            if ($resource !== null) {
+                $resources[$resource->getType()] = $resource;
+            }
+        }
+
+        return $resources;
+    }
+
+    /**
+     * @param Notification $notification
+     * @param NotificationGenericObject[] $users
+     * @param NotificationResource[] $resources
+     * @param NotificationMessage[] $messages
+     *
+     * @return CreatedResponse<int,AddNotificationResponse>
+     */
+    private function createResponse(
+        Notification $notification,
+        array $users,
+        array $resources,
+        array $messages,
+    ): CreatedResponse {
+        $response = new AddNotificationResponse();
+
+        $response->id = $notification->getId();
+        $response->name = $notification->getName();
+        $response->timeperiod = [
+            'id' => $notification->getTimePeriod()->getId(),
+            'name' => $notification->getTimePeriod()->getName(),
+        ];
+        $response->isActivated = $notification->isActivated();
+
+        $response->messages = array_map(
+            (fn($message) => [
+                'channel' => $message->getChannel()->value,
+                'subject' => $message->getSubject(),
+                'message' => $message->getMessage(),
+            ]),
+            $messages
+        );
+
+        $response->users = array_map(
+            (fn($user) => ['id' => $user->getId(), 'name' => $user->getName()]),
+            $users
+        );
+
+        foreach ($resources as $resource) {
+            $eventEnumConverter = $this->resourceRepositoryProvider->getRepository($resource->getType())
+                ->eventEnumConverter();
+            $responseResource = [
+                'type' => $resource->getType(),
+                'events' => $eventEnumConverter::toBitmask($resource->getEvents()),
+                'ids' => array_map(
+                    (fn($resource) => ['id' => $resource->getId(), 'name' => $resource->getName()]),
```

Comment:
put them static when you can :)
```suggestion
                    (static fn($resource) => ['id' => $resource->getId(), 'name' => $resource->getName()]),
```